packages/history-utility/src/index.ts

```diff
@@ -230,7 +233,7 @@ export function proxyWithHistory<V>(
       if (proxyObject.canUndo()) {
         proxyObject.history.wip = proxyObject.clone(
           proxyObject.history.nodes[--proxyObject.history.index]?.snapshot
-        );
+        ) ?? EmptyWIP
```

Comment:
I'm unsure what the expectation is. The issue says that the value is set to undefined and no history change is reflected.. but this is setting the value to an empty object when null or undefined is found.

Seems like a data integrity issue.

What is the expectation here?

Reply:
If I'm understandin correctly, the method `shouldSaveHistory` returns `true` when the value changed _and when the new value is different than the history wip_. 

Since by default (when empty) the history wip is `undefined`, when setting a value to `undefined` the above method returns `false` and the history isn't updated. By setting the default history wip value to an empty object, when the value has changed and the history wip is empty the above method would return `true`, so setting a value to `undefined` would work as expected.

Using an empty object ensures that it is equal (`===`) only to itself, so even setting the new value to an empty object would work as expected, but that is not the use case I was trying to address.

Reply:
Still don't really get the point..if the value is undefined and you try to set it to undefined then there should be no history change..

But if it's set then that means it's not undefined.

And if it sees undefined then undefined will not be equal to wip.. 

So what am I missing?

In theory there is no benefit to changing it to an empty object.

Do your test cases pass without these changes?

Reply:
Thanks.

I'll check it in detail later

Reply:
```
it('should update history when property is set to undefined', async () => {
      const state = proxyWithHistory<
        Partial<{ prop0: number; prop1: number; prop2: number; prop3: number }>
      >({ prop0: 0, prop1: 1, prop2: 2, prop3: 3 });

     // wait for next tick
      await Promise.resolve();

      expect(state.value.prop0).toEqual(0);
      expect(state.value.prop1).toEqual(1);
      expect(state.value.prop2).toEqual(2);
      expect(state.value.prop3).toEqual(3);
      expect(state.history.nodes.length).toEqual(1);

      delete state.value.prop0;

      await Promise.resolve();

      expect(state.value.prop0).toEqual(undefined);
      expect(state.value.prop1).toEqual(1);
      expect(state.value.prop2).toEqual(2);
      expect(state.value.prop3).toEqual(3);
      expect(state.history.nodes.length).toEqual(2);
    });
```

i used this test case without your changes  to understand the issue a little better in isolation of any other changes.

first test case passes and the second one doesnt.

so thanks i have a  clearer understanding

Reply:
Nice, I added your kept your test cases to the vanilla suite and removed mine from react suite.